src/PolarSSL++/RsaPrivateKey.cpp

```diff
@@ -100,7 +100,7 @@ AString cRsaPrivateKey::GetPubKeyDER(void)
 	{
 		return AString();
 	}
-	return AString((const char *)(buf + sizeof(buf) - res), (size_t)res);
+	return AString(static_cast<const char *>(static_cast<const void*>(buf + sizeof(buf) - res)), static_cast<size_t>(res));
```

Comment:
I wasn't aware there was a "weakness" comparison between casts.
In my opinion, `reinterpret_cast` means no change to the data, while `static_cast` means "change the data if needed", so `reinterpret_cast` should be used when converting between raw pointer types.